src/misc/host_locale.h

```diff
@@ -117,14 +110,27 @@ struct HostLocale {
 	// telling which host OS property/value was used to determine the given
 	// locale. Feel free to provide it even if detection failed.
 	struct {
-		std::string keyboard  = {};
 		std::string country   = {};
 		std::string numeric   = {};
 		std::string time_date = {};
 		std::string currency  = {};
 	} log_info = {};
 };
 
+struct HostKeyboardLayouts {
+	// Keyboard layouts, optionally with code pages
+	std::vector<KeyboardLayoutMaybeCodepage> keyboard_layout_list = {};
+
+	// If the keyboard layouts list retrieved from the host OS is already
+	// sorted by user priority, set this to 'true'.
+	bool is_layout_list_sorted = false;
+
+	// If detection was successful, always provide info for the log output,
+	// telling which host OS property/value was used to determine the
+	// language. Feel free to provide it even if detection failed.
```

Comment:
Is this sentence needed/correct?

"Feel free to provide it even if detection failed."

Reply:
Not needed, actually. Removed.